optimum/habana/transformers/generation/utils.py

```diff
@@ -946,6 +944,9 @@ def generate(
             stopping_criteria=stopping_criteria,
             ignore_eos=self.generation_config.ignore_eos,
         )
+        if generation_config.static_shapes and (generation_config.pad_token_id == generation_config.eos_token_id):
+            if any(isinstance(k, EosTokenCriteria) for k in prepared_stopping_criteria):
+                logger.warning(f"For Gaudi, we pad input_ids with pad_token_id (= {generation_config.pad_token_id}), which is equal to eos_token_id for this model, and EosTokenCriteria stopping criteria is requested, so this option (ignore_eos=False) isn't available. Try setting `--ignore_eos` to False.")
```

Comment:
This is a warning for now and not an assert, because some tests like the following will fail otherwise:

`pytest -s -v -k test_generate_from_inputs_embeds_decoder_only test_modeling_gpt2.py`